xtable-core/src/main/java/org/apache/xtable/catalog/TableFormatUtils.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+package org.apache.xtable.catalog;
+
+import static org.apache.iceberg.BaseMetastoreTableOperations.TABLE_TYPE_PROP;
+import static org.apache.xtable.catalog.Constants.PROP_SPARK_SQL_SOURCES_PROVIDER;
+
+import java.util.Map;
+
+import org.apache.iceberg.TableProperties;
+
+import com.google.common.base.Strings;
+
+import org.apache.xtable.exception.NotSupportedException;
+import org.apache.xtable.model.storage.TableFormat;
+
+public class TableFormatUtils {
+
+  public static String getTableDataLocation(
+      String tableFormat, String tableLocation, Map<String, String> properties) {
+    switch (tableFormat) {
+      case TableFormat.ICEBERG:
+        return getIcebergDataLocation(tableLocation, properties);
+      case TableFormat.DELTA:
+      case TableFormat.HUDI:
+        return tableLocation;
+      default:
+        throw new NotSupportedException("Unsupported table format: " + tableFormat);
+    }
+  }
+
+  /** Get iceberg table data files location */
+  private static String getIcebergDataLocation(
+      String tableLocation, Map<String, String> properties) {
+    String dataLocation = properties.get(TableProperties.WRITE_DATA_LOCATION);
+    if (dataLocation == null) {
+      dataLocation = properties.get(TableProperties.WRITE_FOLDER_STORAGE_LOCATION);
+      if (dataLocation == null) {
+        dataLocation = properties.get(TableProperties.OBJECT_STORE_PATH);
+        if (dataLocation == null) {
+          dataLocation = String.format("%s/data", tableLocation);
+        }
+      }
+    }
+    return dataLocation;
+  }
+
+  // Get table format name from table properties
```

Comment:
Use proper javadocs here please. 

Also will all catalogs be required to set this property?

Reply:
Yes. Btw, this property should be set automatically by the engine (which created the table) in most of the cases

Does this javadoc make sense - 

<img width="687" alt="image" src="https://github.com/user-attachments/assets/11338790-5d47-45ed-b841-483d0bb5eb4c" />

Reply:
Why will the spark type be set? There is no guarantee spark created the table. 
What should the code do if tableFormat is still null at the end of the method.

Reply:
- As per the documentation: https://docs.aws.amazon.com/athena/latest/ug/delta-lake-tables-syncing-metadata.html, Athena and AWS Glue consoles requires `spark.sql.sources.provider` property to be set when creating delta lake table using those methods.

- We will throw error if unable to get tableFormat as this is needed to create ConversionSource from SourceTable: https://github.com/apache/incubator-xtable/pull/599/files#diff-90350a03dbbf828930b2ad5b2629bb8c9da7aaa24b7f984600e03e807a185f7bR79

Reply:
Does it make more sense to move that logic into the method instead of making each caller throw that exception?